redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/EventListener.kt

```diff
@@ -148,6 +150,16 @@ public abstract class EventListener {
   ) {
   }
 
+  /** Invoked on a request to process an unknown event [tag] for the specified widget [widgetTag]. */
+  public open fun onUnknownEvent(widgetTag: WidgetTag, tag: EventTag) {
+    println("Unknown event tag ${tag.value} for widget tag ${widgetTag.value}")
```

Comment:
None of the other callbacks log by default. These should probably be empty implementations to match.

Reply:
I think these are needed

This work started (at least how I know it, @swankjesse probably has a bigger picture story) because Jesse and I were looking at the emoji picker on Activity (in treehouse), and the app crashes because of one of these two events (from the default [protocol mismatch handler](https://github.com/cashapp/redwood/blob/2bb11847d890d2c89b57dd8fa285c4bb30532c54/redwood-protocol-compose/src/commonMain/kotlin/app/cash/redwood/protocol/compose/ProtocolMismatchHandler.kt#L38)), with this change the same case won't crash anymore, but I also would like to see a log when it happens (instead of being silent)

Reply:
Writing to stdout via `println` or any other means is basically a blackhole in production. We have no visibility into these logs and will never see them anywhere but on the emulator. We _must_ override these functions and send the logs to our own logging wrapper (which will eventually forward to the same place stdout goes) or report through some other means in order to have them be visible in the wild.

Conversely, writing to stdout automatically is lightly frowned upon in production because it adds noise to the global logs. In production our logging wrapper does not write to stdout but only captures logs and sends them to the server.

Reply:
ya right now things are in dev stage, I think we'll need a different plan for prod (like writing to bugsnag/emitting events etc.)

Reply:
Definitely. And the way we'll do that is with some other subclass of this event listener specific to our needs.

I guess what I'm saying is that none of the other callbacks in this listener call `println`. So either we should have them all call `println` and the default behavior is to log, or none of them should and the default behavior is to do nothing. I personally lean very heavily towards doing nothing because we can't assume anything about the environment in which the user is running Treehouse.

Reply:
Also I just paired with @swankjesse and he showed me how Android repo is handling this. Removed!

Reply:
Thanks!